### ISSUE TITLE
[master] Assertion failure when running in Debug mode

### ISSUE THREAD

**Issue by zappybiby** [bethesirius/ChosunTruck#22]
This error occurs at startup while in Debug mode:

`OpenCV Error: Assertion failed (src.checkVector(2, CV_32F) == 4 && dst.checkVect or(2, CV_32F) == 4) in cv::getPerspectiveTransform, file ...\opencv_sources\modules\imgproc\src\imgwarp.cpp, line 6353`

This makes the program not run in Debug mode. This is a big problem for me, as Debug mode would allow for better debugging.

The program runs fine in Release mode.

**Comment by zappybiby** [comment]
Looking at some forums, this is caused by the wrong file type being used in the transformation. 

See http://answers.opencv.org/question/18252/opencv-assertion-failed-for-perspective-transform/

**Comment by simjaecheol** [comment]
Have you tried this one?
http://stackoverflow.com/questions/15683184/assertion-failed-when-im-trying-to-use-getperspectivetransform-on-android-ndk-t
This is android opencv problems, but I think it looks very similar to your problem.
I hope it works. :)

**Comment by zappybiby** [comment]
Where do we define the type of the mat in our master branch?

**Comment by simjaecheol** [comment]
In IPM::createMaps() in IPM.cpp, there are 

m_mapX.create(m_dstSize, CV_32F);
m_mapY.create(m_dstSize, CV_32F);
...
m_invMapX.create(m_origSize, CV_32F);
m_invMapY.create(m_origSize, CV_32F);

How about change CV_32F to CV_32FC2?

**Comment by zappybiby** [comment]
Still the same error :( 

I have no idea

**Comment by zappybiby** [comment]
and the vector size of each Size + points is 4, so it should work anyway.

**Comment by zappybiby** [comment]
CV_32FC2 actually makes Release fail to work.

_origSize,  _dstSize, _origPoints, _dstPoints are all size 4 according to the debugger. Not sure why its failing the assertion.

**Comment by simjaecheol** [comment]
Looking at CvType, CV_32FC2 contains x and y coordinates. ([reference](http://stackoverflow.com/questions/34454178/convert-a-matrix-of-type-cv-32fc2-to-type-cv-32fc1))
In our case, however, only one coordinate is used. 

float* ptRowX = m_mapX.ptr<float>(j);
...
ptRowX[i] = pt.x;

For this reason, I think there is an error in release mode.

* CV_32F defines the depth of each element of the matrix, while
* CV_32FC# defines both the depth of each element and the number of channels.
[Difference between OpenCV type CV_32F and CV_32FC1](http://stackoverflow.com/questions/37530368/difference-between-opencv-type-cv-32f-and-cv-32fc1)

I think CV_32FC1 is suitable in our case.

**Comment by chi3236** [comment]
This is a little off-topic opinion, but if you execute the program which is built with debug mode, it will be REALLY slow. 

At the [early time of the project](https://github.com/bethesirius/ChosunTruck/blob/5883fd9184b6e952972f938211fcdf51fe1407f5/ets2_auto_driving/main.cpp), I built the program with debug mode. Although the program did not have the steering function, I got less than 10 fps with Intel Haswell i7 processor.
